### PR TITLE
Us003 view capability

### PR DESCRIPTION
**Pull request by ConnMeehan**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+var express = require('express');
```
Comment (kamil-biegaj) [inline app.js]: Don't need this, we are using TS

File: controller/JobRoleController.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
+"use strict";
```
Comment (kamil-biegaj) [inline controller/JobRoleController.js]: Dont need the "use strict"

File: controller/JobRoleController.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+module.exports = function(app: Application) {
+    app.get('/capability', async (req, res) => {
+        let data = [];
+
+        try {
+                data = await jobRoleService.getCapability() 
+            } catch (e) {
+                console.error(e);
```
Comment (kamil-biegaj) [inline controller/JobRoleController.ts:27]: Set an actual error message to show in UI

File: model/Capability.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+"use strict";
```
Comment (kamil-biegaj) [inline model/Capability.js]: Same thing as before, only commit TS

File: model/Capability.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+export class Capability {
```
Comment (kamil-biegaj) [inline model/Capability.ts]: Do these need to be optional, the (?)

File: service/JobRoleService.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+        const response = await axios.get('http://localhost:8080/api/job_roles')
+        return response.data
+    } catch (e) {
+        throw new Error('Could not get job roles.')
+    }
+}
+
+module.exports.getCapability = async function() {
+    try {
+        const response = await axios.get('http://localhost:8080/api/capability')
```
Comment (kamil-biegaj) [inline service/JobRoleService.ts]: Use environment variable, not hardcode the server

File: service/JobRoleService.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        return response.data
+    } catch (e) {
+        throw new Error('Could not get capabilities.')
```
Comment (kamil-biegaj) [inline service/JobRoleService.ts]: Log out the error to help with debugging

File: test/unit/service/JobRoleServiceTest.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            mock.onGet(JobRoleService.URL).reply(200, data);
+    
+            var results = await JobRoleService.getJobRoles();
+    
+            expect(results[0]).to.deep.equal(jobrole)
+          })
+    })
+})
+
+Test
```
Comment (kamil-biegaj) [inline test/unit/service/JobRoleServiceTest.js]: Need a failed test, to show axios returning 500

**Review by kamil-biegaj — CHANGES_REQUESTED** [review]

**Review by kamil-biegaj — APPROVED** [review]

**Review by AlexMcC02 — APPROVED** [review]
Looks fine to me.

File: views/list-capabilities.html [hunk 9]
```diff
@@ -0,0 +1,54 @@
+<!DOCTYPE html>
```
Comment (shaunganley) [inline views/list-capabilities.html]: you don't need a new view, update the existing view to show the capability

File: service/JobRoleService.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+module.exports.getCapability = async function() {
+    try {
+        const response = await axios.get(process.env.UI_URL + '/api/capability')
+        return response.data
+    } catch (e) {
+        console.log('Could not get capabilities.')
+        throw new Error('Could not get capabilities.')
+    }
```
Comment (shaunganley) [inline service/JobRoleService.ts]: not needed

File: controller/JobRoleController.ts [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+                res.locals.errormessage = "Failed to fetch capabilities"
+                return res.render('list-capabilities', req.body);
+            }
+            res.render('list-capabilities', { capabilities: data } )
+    })
```
Comment (shaunganley) [inline controller/JobRoleController.ts]: not needed

**Review by shaunganley — CHANGES_REQUESTED** [review]
you don't need a new page to display the capability name, just update the existing one

File: middleware/auth.ts [hunk 12]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (shaunganley) [inline middleware/auth.ts]: not needed yet, delete

File: views/list-jobroles.html [hunk 13]
```diff
@@ -33,13 +33,15 @@ <h2 id = "jobs">Jobs</h2>
                     <tr>
                         <th scope="col" id = "name">Name</th>
                         <th scope="col" id = "specification">Specification</th>
+                        <th scope="col" id = "specification">Capability</th>
```
Comment (shaunganley) [inline views/list-jobroles.html]: incorrect id

**Review by shaunganley — APPROVED** [review]

**Pull request merged**